utils/parse.go

```diff
@@ -31,6 +31,14 @@ func ParseShell(input string) []string {
 
 	args := strings.FieldsFunc(input, f)
 	for i := 0; i < len(args); i++ {
+		quotationMarks := regexp.MustCompile(`"`)
```

Comment:
We should update this function to return `([]string, error)`.

A faster way to do this would be to count marks as we iterate through `FieldsFunc`:
```go
count := 0
lastQuote := rune(0)
f := func(r rune) bool {
    switch {
    ...
    case unicode.Is(unicode.Quotation_Mark, r):
        if r == '"' {
            count++
        }
        ...
}

args := strings.FieldsFunc(input, f)
if count % 2 != 0 {
    return nil, fmt.Errorf("...")
}

for i := 0; i < len(args); i++ {
...
```